numpy/core/tests/test_function_base.py

```diff
@@ -407,3 +407,29 @@ def test_round_negative(self):
         y = linspace(-1, 3, num=8, dtype=int)
         t = array([-1, -1, 0, 0, 1, 1, 2, 3], dtype=int)
         assert_array_equal(y, t)
+
+
+class TestArange:
+    def test_require_range(self):
+        import pytest
+        with pytest.raises(TypeError):
+            arange()
+
+        with pytest.raises(TypeError):
+            arange(step=3)
+
+        with pytest.raises(TypeError):
+            arange(dtype='int16')
+
+        with pytest.raises(TypeError):
+            arange(start=4)
+
+    def test_start_stop_kwarg(self):
+        keyword_stop = arange(stop=3)
+        keyword_zerotostop = arange(start=0, stop=3)
+        keyword_start_stop = arange(start=3, stop=9)
+
+        assert len(keyword_stop) == 3
+        assert len(keyword_zerotostop) == 3
+        assert (keyword_stop == keyword_zerotostop).all()
```

Comment:
Just FYI, we also have a `assert_arrays_equal`, which checks also shape/dtype (difference that `all()` might miss). It doesn't matter here though.